apps/chat/src/components/Chat/Publish/TargetAudienceFilterComponent.tsx

```diff
@@ -99,62 +127,143 @@ export function TargetAudienceFilterComponent({
     [],
   );
 
-  const isSaveBtnDisabled =
-    selectedTarget === emptySelector || // a target is not selected
-    filterFunction === emptySelector || // a filter function is not selected
-    (!filterParams.length && !filterRegexParam) || // filter params are not selected and regex hasn't filled
-    !!(
-      // regex filled, but the regex filter function isn't selected
-      (
-        filterRegexParam &&
-        filterFunction !== PublicationFunctions.Regex &&
-        !filterParams.length
-      )
-    ) ||
-    !!(
-      // params selected, but the regex filter function isn't selected
-      (
-        filterParams.length &&
-        filterFunction === PublicationFunctions.Regex &&
-        !filterRegexParam
-      )
+  const isTargetAndFunctionSelected =
+    selectedTarget !== emptySelector && filterFunction !== emptySelector;
+  const areSomeFilterParamSelected = filterParams.length || filterRegexParam;
+  const isRegexFilledInButNotSelected = !!(
+    filterRegexParam &&
+    filterFunction !== PublicationFunctions.Regex &&
+    !filterParams.length
+  );
+  const isParamsFilledInButRegexIsSelected = !!(
+    filterParams.length &&
+    filterFunction === PublicationFunctions.Regex &&
+    !filterRegexParam
+  );
+  const isTrueOrFalseFilterSelected =
+    filterFunction === PublicationFunctions.True ||
+    filterFunction === PublicationFunctions.False;
```

Comment:
I'm not sure if good idea, but we can create help-method
```
const isTrueOrFalseFilter = (filterFunction) =>
    filterFunction === PublicationFunctions.True ||
    filterFunction === PublicationFunctions.False;
```
and reuse in several places

Reply:
I think it will be code for code's sake. Right now it's only needed for this particular case.